### PR TITLE
 plan: no need to double read in some case.

### PR DESCRIPTION
**Pull request by winoros**
Since index of string type may have length currently, consider one sql like the following:
```sql
create table t(id varchar(50) primary key);
select * from t where id > 'aaa';
```
The select statement will be double read, which single read is enough.

### CODE REVIEW

**Comment by winoros** [comment]
PTAL @coocood @hanfei1991 @zz @shenli

File: plan/physical_plan_builder.go [hunk 1]
```diff
@@ -691,7 +691,8 @@ func isCoveringIndex(columns []*model.ColumnInfo, indexColumns []*model.IndexCol
 		}
 		isIndexColumn := false
 		for _, indexCol := range indexColumns {
-			if colInfo.Name.L == indexCol.Name.L && indexCol.Length == types.UnspecifiedLength {
+			isUnspecifiedLen := indexCol.Length == types.UnspecifiedLength || indexCol.Length == colInfo.Flen
```
Comment (coocood) [inline plan/physical_plan_builder.go]: `isFullLen` is a better name.

**Comment by shenli** [comment]
LGTM

**Comment by coocood** [comment]
Any test to cover it?

**Comment by winoros** [comment]
@coocood I'm finding somewhere to add it.

**Comment by winoros** [comment]
@coocood A little change in `ranger_test.go` to cover this.

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**